.github/workflows/release-linux.yaml

```diff
@@ -0,0 +1,103 @@
+name: Release Linux
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  cuda-build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        cuda: ['11.7.0', '12.1.0']
+    steps:
+      - uses: jimver/cuda-toolkit@v0.2.11
+        with:
+          cuda: ${{ matrix.cuda }}
+          method: network
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21
+
+      - name: Install Linux build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake
+
+      - name: Generate CUDA runner
+        run: |
+          go generate ./...
+
+      - name: Prepare CUDA output for upload
+        run: |
+          tar -czvf artifact-${{ matrix.cuda }}.tar.gz ./llm/llama.cpp
```

Comment:
Why create an archive instead of uploading the entire build directory?

Reply:
When I upload the build directory github automatically zips it, which causes the llama.cpp server executable to lose its executable bit and no longer be seen as an file which can be run.

Reply:
That's simple enough to fix and a lot less complex than untaring these archives